dev-embedded/jal/files/jal-0.4.62-fix-incompatible-integer-to-pointer-clang16.patch

```diff
@@ -0,0 +1,24 @@
+Bug: https://bugs.gentoo.org/883243
+--- a/jal/treerep.c
++++ b/jal/treerep.c
+@@ -892,7 +892,7 @@ tree new_asm(loc_t loc, opcode_t opcode, tree arg1, int arg2)
+ 
+ 
+     if (code_has(p->opcode, field_flabel)) {
+-        assert_pointer(NULL, arg2);
++        assert_pointer(NULL, &arg2);
```

Comment:
This is wrong.  By looking at the code, `assert_pointer` seems to do some pointer arithmetic to determine if the given value is inside some custom allocation block; `&arg2` points to the stack, so the assert will always fail.  I suppose it expects arg2 to be a pointer as int, so a simple cast should be sufficient in this case.